### PR TITLE
SLVSCODE-120 notify user if NodeCommandException

### PR DESCRIPTION
**Pull request by kirill-knize-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarsource/sonarlint/ls/log/LanguageClientLogOutput.java [hunk 1]
```diff
@@ -73,6 +77,15 @@ public LanguageClientLogOutput(LanguageClient client) {
 
   @Override
   public void log(String formattedMessage, Level level) {
+    if (formattedMessage.contains("NodeCommandException")) {
+      MessageActionItem actionItem = new MessageActionItem("Show SonarLint Output");
+      ArrayList<MessageActionItem> actionItems = new ArrayList<>();
+      actionItems.add(actionItem);
+      ShowMessageRequestParams params = new ShowMessageRequestParams(actionItems);
+      params.setType(MessageType.Error);
+      params.setMessage("JS/TS analyzer issue. Look at the log output for more details.");
```
Comment (jblievremont) [inline src/main/java/org/sonarsource/sonarlint/ls/log/LanguageClientLogOutput.java:86]:
```suggestion
      params.setMessage("JS/TS analysis failed. Please check the SonarLint Output for more details.");
```

**Review by jblievremont — CHANGES_REQUESTED** [review]
Almost there! Please see my comments.

File: src/main/java/org/sonarsource/sonarlint/ls/log/LanguageClientLogOutput.java [hunk 2]
```diff
@@ -73,6 +77,15 @@ public LanguageClientLogOutput(LanguageClient client) {
 
   @Override
   public void log(String formattedMessage, Level level) {
+    if (formattedMessage.contains("NodeCommandException")) {
+      MessageActionItem actionItem = new MessageActionItem("Show SonarLint Output");
+      ArrayList<MessageActionItem> actionItems = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (kirill-knize-sonarsource) [reply src/main/java/org/sonarsource/sonarlint/ls/log/LanguageClientLogOutput.java]: Done.

File: src/test/java/org/sonarsource/sonarlint/ls/log/LanguageClientLogOutputTests.java [hunk 3]
```diff
@@ -125,4 +131,24 @@ public void enable_analyzer_debug_logs() {
     verifyNoMoreInteractions(languageClient);
   }
 
+  @Test
+  public void notification_to_client_for_node_command_exception() {
+
+    MessageActionItem actionItem = new MessageActionItem("Show SonarLint Output");
```
[the remark this answers is not in the document]
Comment (kirill-knize-sonarsource) [reply src/test/java/org/sonarsource/sonarlint/ls/log/LanguageClientLogOutputTests.java]: Done.

File: src/main/java/org/sonarsource/sonarlint/ls/log/LanguageClientLogOutput.java (hunk 1 above)
Comment (kirill-knize-sonarsource) [reply src/main/java/org/sonarsource/sonarlint/ls/log/LanguageClientLogOutput.java]: Done.

File: src/test/java/org/sonarsource/sonarlint/ls/mediumtests/AbstractLanguageServerMediumTests.java [hunk 4]
```diff
@@ -296,6 +296,11 @@ public void logMessage(MessageParams message) {
       });
     }
 
+    @Override
+    public CompletableFuture<Void> showSonarLintOutput() {
+      return null;
```
[the remark this answers is not in the document]
Comment (kirill-knize-sonarsource) [reply src/test/java/org/sonarsource/sonarlint/ls/mediumtests/AbstractLanguageServerMediumTests.java:301]: Done.

**Review by jblievremont — APPROVED** [review]
LGTM

**Pull request merged**